test/Service/SessionManagerFactoryTest.php

```diff
@@ -84,12 +104,19 @@ public function testFactoryWillAddValidatorViaConfiguration()
                 'Zend\Session\Validator\RemoteAddr',
             ],
         ]];
-        $this->services->setService('Config', $config);
-        $manager = $this->services->get('Zend\Session\ManagerInterface');
+        $services = $this->services->withConfig([
+            'services' => [
+                'config' => $config,
+            ],
+        ]);
+        $manager = $services->get('Zend\Session\ManagerInterface');
 
         $manager->start();
 
-        $this->assertEquals(1, $manager->getValidatorChain()->getListeners('session.validate')->count());
+        $chain = $manager->getValidatorChain();
+        $r = new \ReflectionMethod($chain, 'getListenersByEventName');
```

Comment:
@Maks3w - for tests, I've been using a similar pattern in other components.
The getListeners and getEvents methods only really existed for testing
purposes, which is why they were removed during the refactor. The downside
is, of course, that testing now often needs to rely on reflection to
validate expectations. My point is: there isn't a way currently to verify
using the public API at this point other than triggering the event, which
often has side effects. Reflection should be fine here.
On Oct 25, 2015 2:09 PM, "Maks3w" <notifications@github.com> wrote:

> In test/Service/SessionManagerFactoryTest.php
> <https://github.com/zendframework/zend-session/pull/8#discussion_r42948778>
> :
>
> >
> >          $manager->start();
> >
> > -        $this->assertEquals(1, $manager->getValidatorChain()->getListeners('session.validate')->count());
> > +        $chain = $manager->getValidatorChain();
> > +        $r = new \ReflectionMethod($chain, 'getListenersByEventName');
>
> I would like to see an alternative to this using the public API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-session/pull/8/files#r42948778>.
>